beekeeper-path-scheduler-apiary/src/main/java/com/expediagroup/beekeeper/scheduler/apiary/filter/MetadataOnlyListenerEventFilter.java

```diff
@@ -27,6 +27,9 @@
 
   @Override
   public boolean filter(ListenerEvent listenerEvent) {
+    if (listenerEvent == null) {
```

Comment:
When does this happen?

Reply:
This could happen if there's some problem with message desrialisation. It _probably_ wont ever happen.

We currently only have this check in `EventTypeListenerEventFilter`. All the filters are gathered in a list and executed one by one. This means that the `EventTypeListenerEventFilter` currently has to be first - and we've got no unit test for that. So I thought I'd add this check to all of the filters, and not have to rely on the ordering of them to catch edge cases.

Reply:
Could we not do the null check in FilteringMessageReader and not call the Filters if its null? i.e. guarantee the filters that we'll never call them with null? I don't really mind and I see the value of having a check here but on the other hand I don't like that we now have to remember to put this null check into every single filter - too much repeated code.